tests/ui/traits/bound/not-on-bare-trait.stderr

```diff
@@ -26,9 +34,26 @@ LL | fn foo(_x: impl Foo + Send) {
    |            ++++
 help: function arguments must have a statically known size, borrowed types always have a known size
    |
-LL | fn foo(_x: &Foo + Send) {
+LL | fn foo(_x: &(Foo + Send)) {
+   |            ++          +
+
+error[E0277]: the size for values of type `(dyn Foo + Send + 'static)` cannot be known at compilation time
+  --> $DIR/not-on-bare-trait.rs:12:8
+   |
+LL | fn bar(_x: (dyn Foo + Send)) {
+   |        ^^ doesn't have a size known at compile-time
+   |
+   = help: the trait `Sized` is not implemented for `(dyn Foo + Send + 'static)`
+   = help: unsized fn params are gated as an unstable feature
+help: you can use `impl Trait` as the argument type
+   |
+LL | fn bar(_x: impl (dyn Foo + Send)) {
+   |            ++++
```

Comment:
rats! Given that we're doing some sketchy string handling already, I don't think this is the only problem we'll face, but I'd rather not add *more* string handling logic to this.

Reply:
You have access to the `self_ty: hir::Ty<'tcx>` -- why not just check the `TraitObjectSyntax` here? we shouldn't need any string manipulation here at all.

Reply:
This one is from the second commit, not the first one. That one only has the additional `Span` from the `ObligationCauseCode::SizedArgumentType`. We could change that one to instead hold the `HirId` instead... 🤔

Reply:
Done.